src/realm/impl/transact_log.hpp

```diff
@@ -1133,6 +1148,11 @@ class TransactReverser {
         return true;
     }
 
+    bool dictionary_clear(size_t)
+    {
+        return true;
```

Comment:
I think this actually needs to call `m_encoder.dictionary_insert()`, see list_clear() and set_clear().
IIRC, this affects notifications when rolling back a write transaction that has cleared a dictionary. Not sure what keys would be used here because we have no record of them though. It would be good to have a test to cover this case.

Reply:
Notifications for rollbacks are only applicable to KVO, which doesn't report any information for changes on dictionaries beyond that the dictionary changed in some way (unlike arrays, where we report the indices that changed). This means we need to generate some instruction that touches the dictionary, but which one is unimportant.

Reply:
OK, it is a little bit counterintuitive that we have to issue some `insert instruction` while clearing the collection, but I guess since it is used only for notifying that the dictionary has changed we could just do something like `insert(pos, Mixed{})`

Reply:
Thanks for the clarification Thomas. 
@nicola-cab this is for the `TransactReverser` which undoes instructions during a rollback. As Thomas says, we only need one modification for the notification system to pick this up, so you only need one insert here. Inserting the same key many times in a loop is a no-op because a dictionary can only have one key of the same value. I'd say its worth adding a comment about this discussion there though.

Reply:
You can follow the test `LangBindHelper_RollbackAndContinueAsRead_TransactLog` for inspiration in covering this case.